subprojects/docs/src/snippets/files/fileCollections/kotlin/build.gradle.kts

```diff
@@ -18,27 +18,29 @@ file("src2/dir2").mkdirs()
 
 // tag::closure[]
 tasks.register("list") {
+    val projectDirectory: Directory = layout.projectDirectory
     doLast {
         var srcDir: File? = null
 
-        val collection = layout.files({
+        val collection = projectDirectory.files({
             srcDir?.listFiles()
         })
 
-        srcDir = file("src")
+        srcDir = projectDirectory.file("src").asFile
         println("Contents of ${srcDir.name}")
-        collection.map { relativePath(it) }.sorted().forEach { println(it) }
+        collection.map { it.relativeTo(projectDirectory.asFile) }.sorted().forEach { println(it) }
 
-        srcDir = file("src2")
+        srcDir = projectDirectory.file("src2").asFile
         println("Contents of ${srcDir.name}")
-        collection.map { relativePath(it) }.sorted().forEach { println(it) }
+        collection.map { it.relativeTo(projectDirectory.asFile) }.sorted().forEach { println(it) }
     }
 }
 // end::closure[]
 
 tasks.register("usage") {
+    val projectLayout: ProjectLayout = layout
```

Comment:
🔴 Prefer to omit the type declaration in Kotlin unless it's necessary for correctness.

Reply:
Done 👍